maestro-studio/web/src/InteractPage.tsx

```diff
@@ -0,0 +1,43 @@
+import InteractableDevice from './InteractableDevice';
+import ReplView from './ReplView';
+import React, { useEffect, useState } from 'react';
+import { DeviceScreen } from './models';
+import { API, wait } from './api';
+
+const InteractPage = () => {
+  const [deviceScreen, setDeviceScreen] = useState<DeviceScreen>()
+
+  useEffect(() => {
+    let running = true;
+    (async () => {
+      while (running) {
+        try {
+          const deviceScreen = await API.getDeviceScreen(true)
+          setDeviceScreen(deviceScreen)
+        } catch (e) {
+          console.error(e)
+          await wait(1000)
+        }
+      }
+    })()
+    return () => { running = false }
+  }, [setDeviceScreen])
+
+  if (!deviceScreen) return null
```

Comment:
Do we want a fallback message here when there is no device screen available?

Reply:
Yeah error / loading states, etc are still wip